client/my-sites/checkout/checkout-thank-you/index.jsx

```diff
@@ -267,6 +268,17 @@ export class CheckoutThankYou extends React.Component {
 		}
 	};
 
+	redirectIfDomainOnly = props => {
+		if ( props.domainOnlySiteFlow && get( props, 'receipt.hasLoadedFromServer', false ) ) {
+			const purchases = getPurchases( props );
+			const failedPurchases = getFailedPurchases( props );
+			if ( props.domainOnlySiteFlow && purchases.length > 0 && ! failedPurchases.length ) {
```

Comment:
We don't need to check `props.domainOnlySiteFlow` since we already know that it is `true` according to the above `if` clause.